crates/runtime/tests/kafka/mod.rs

```diff
@@ -39,19 +39,34 @@ async fn kafka_sasl_connect_test() -> anyhow::Result<()> {
     test_request_context()
         .scope(async {
             let (running_container, producer) =
-                start_kafka_docker_container(KAFKA_PORT, "orders").await?;
+                start_kafka_docker_container(KAFKA_PORT, &["orders", "schema_infer_test"]).await?;
 
             tracing::debug!("Container started");
 
+            // Load test data for orders representing the simple case where all fields are present in the first topic message
             let orders_simple: Vec<serde_json::Value> =
                 serde_json::from_str(include_str!("./test_data/orders_simple.json"))?;
-
             send_messages_to_kafka(&producer, "orders", &orders_simple).await?;
 
-            let ds = make_kafka_dataset("orders", "kafka_orders", KAFKA_PORT);
+            // Load test data for orders representing a more complex schema inference case where
+            // the first messages do not have all fields present and some contain nulls
+            let orders_schema_infer: Vec<serde_json::Value> =
+                serde_json::from_str(include_str!("./test_data/orders_schema_infer.json"))?;
+            send_messages_to_kafka(&producer, "schema_infer_test", &orders_schema_infer).await?;
+
+            let ds = make_kafka_dataset("orders", "kafka_orders", KAFKA_PORT, None);
+            let mut options = std::collections::HashMap::new();
```

Comment:
nit: You can oneline these if you want

```suggestion
            let options: std::collections::HashMap = [("kafka_schema_infer_num_samples".to_string(), "3".to_string())].into();
```
```